### ISSUE TITLE
different shaping result with DirectWrite for an Arabic Quranic word

### ISSUE THREAD

**Issue by aminabedi68** [harfbuzz/harfbuzz#3530]
I was checking Quran with Calibri font and find a different shaping result:
HarfBuzz(firefox98, chrome101) and DirectWrite(Wordpad and Ms Office 2021 on windows 10) rendering different shapes for these sequence of codepoints in Calibri font:
`0634 064E 0646 064E 200D 0654 064E 0627 0646 064F`

HarfBuzz result:
![firefox_Syv8gfKZzu](https://user-images.githubusercontent.com/25493297/162216383-d6219a78-c40d-4ed1-8d91-da7fa0a07ec1.png)
DirectWrite result(which seems correct):
![wordpad_FGYWYz4rju](https://user-images.githubusercontent.com/25493297/162216406-8accd2d8-c136-45a3-90df-72db0eba2a34.png)
for Harbuzz it seems that the `rclt` feature(Calibri font GSUB number 114) not working with `200D`.

**Comment by khaledhosny** [comment]
That is a peculiar way to encode this string. I would have expected U+0640 (tatweel) in place of that U+200D (ZWJ). What happens if you use U+034F (CGJ) instead?

**Comment by aminabedi68** [comment]
yes, i agree. it makes more sense to use tatweel. the text and font aren't mine, i just saw it and realized that text sequence predicted in opentype features of the font, so maybe it's an old routine but expected to working.

**Comment by aminabedi68** [comment]
oh! it's more confusing now! here is another word with similar sequence but working this time!
`0628 0650 200D 0654 064E 0627 064A 064E 0670 062A 0650 064A`
![firefox_aD3voJcYLS](https://user-images.githubusercontent.com/25493297/162310882-3422182a-4fc6-45f5-81b6-345884d609fa.png)

**Comment by behdad** [comment]
@khaledhosny Do you know which font this is? I don't think Calibri supports Arabic.

**Comment by khaledhosny** [comment]
Recent-ish versions of Calibri support Arabic.

**Comment by behdad** [comment]
> Recent-ish versions of Calibri support Arabic.

Okay I don't have access to that.
